### PR TITLE
[trend_micro_vision_one] - Update package-spec to 2.9.0

### PR DESCRIPTION
**Pull request by taylor-swanson**
# What does this PR do?

- Updates the package-spec version to `2.9.0`
- Ensures `source.user.roles` and `observer.mac` fields are arrays

```
[git-generate]
go run github.com/andrewkroh/go-examples/ecs-update@latest -ecs-version=8.8.0 -ecs-git-ref=8.8 -format-version=2.9.0 packages/trend_micro_vision_one
```

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates elastic/security-team#5870

### CODE REVIEW

**Comment by taylor-swanson** [comment]
Looks like elastic-package has not been updated yet, waiting on #6679

File: packages/trend_micro_vision_one/data_stream/audit/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -97,10 +97,10 @@ processors:
       field: json.loggedRole
       target_field: trend_micro_vision_one.audit.logged_role
       ignore_missing: true
-  - set:
+  - append:
       field: source.user.roles
-      copy_from: trend_micro_vision_one.audit.logged_role
-      ignore_failure: true
+      value: '{{{trend_micro_vision_one.audit.logged_role}}}'
+      if: ctx.source?.user?.roles != null
```
Comment (efd6) [inline packages/trend_micro_vision_one/data_stream/audit/elasticsearch/ingest_pipeline/default.yml:103]:
```suggestion
      if: ctx.trend_micro_vision_one?.audit?.logged_role != null
```
?

Comment (taylor-swanson) [reply packages/trend_micro_vision_one/data_stream/audit/elasticsearch/ingest_pipeline/default.yml:103]: Oops, must've been a copy/paste error and didn't notice. Will fix...

**Review by efd6 — APPROVED** [review]

**Pull request merged**